### PR TITLE
Hard Limit in Config (for Pagination)

### PR DESCRIPTION
**Pull request by saaiful**
https://github.com/yajra/laravel-datatables/issues/2493

### CODE REVIEW

File: src/config/datatables.php [hunk 1]
```diff
@@ -119,4 +119,10 @@
         'options' => 0,
     ],
 
+    /**
+     * Maximum record per page
```
Comment (yajra) [inline src/config/datatables.php:123]: I think the correct word is records?

`Maximum records per page.`

**Review by yajra — CHANGES_REQUESTED** [review]

**Comment by yajra** [comment]
PR looks good aside from the minor typo? Will test this today on actual project. Thanks!

**Comment by yajra** [comment]
Also, maybe we can also include `records_per_page` on the config while we're at it instead of hard coding it at 10?

**Comment by yajra** [comment]
Just tested this and it breaks the dataTables info? Tried setting the max records to 3 and the requested length is 10.

image

Expected info should be something like  `Showing 1 to 3 of 10 results` or I might be missing something?

**Comment by saaiful** [comment]
> Just tested this and it breaks the dataTables info? Tried setting the max records to 3 and the requested length is 10.
> 
> image
> 
> Expected info should be something like `Showing 1 to 3 of 10 results` or I might be missing something?

I will try to fix that.

**Comment by saaiful** [comment]
I just solved the issue.
In Yajra\DataTables\Html\Builder
```/**
     * Generate DataTables js parameters.
     *
     * @param  array $attributes
     * @return string
     */
    public function parameterize($attributes = [])
    {
        $parameters = (new Parameters($attributes))->toArray();

        $max_record_per_page = $this->config->get('datatables.max_record_per_page', 0);
        if (!array_key_exists('lengthMenu', $parameters)) {
            $parameters['lengthMenu'] = [10, 25, 50, 100];
        }

        if ($max_record_per_page != 0) {
            $lengthMenu = array_unique($parameters['lengthMenu']);
            foreach ($lengthMenu as $key => $value) {
                if ($value > $max_record_per_page) {
                    unset($lengthMenu[$key]);
                }
            }
            $lengthMenu = array_values($lengthMenu);
            if (empty($lengthMenu)) {$lengthMenu = [$max_record_per_page];}
            sort($lengthMenu);
            $parameters['lengthMenu'] = $lengthMenu;
        }

        $values = [];
        $replacements = [];

        foreach (Arr::dot($parameters) as $key => $value) {
            if ($this->isCallbackFunction($value, $key)) {
                $values[] = trim($value);
                Arr::set($parameters, $key, '%' . $key . '%');
                $replacements[] = '"%' . $key . '%"';
            }
        }

        $new = [];
        foreach ($parameters as $key => $value) {
            Arr::set($new, $key, $value);
        }

        $json = json_encode($new);

        $json = str_replace($replacements, $values, $json);

        return $json;
    }
```

This will insert `lengthMenu` parameter according to `max_record_per_page` in config file. This will not affect manual `lengthMenu` insertion in "builder parameters".

**Pull request closed without merging**